server/monitor-types/snmp.js

```diff
@@ -0,0 +1,102 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN, log, getKey } = require("../../src/util");
+const snmp = require("net-snmp");
+
+class SNMPMonitorType extends MonitorType {
+    name = "snmp";
+
+    /**
+     * @inheritdoc
+     */
+    async check(monitor, heartbeat, _server) {
+
+        const options = {
+            port: monitor.port || "161",
+            retries: monitor.maxretries,
+            timeout: monitor.timeout * 1000,
+
+            // Resolve the net-snmp version identifier from monitor.snmpVersion using the `getKey` method (see utils.ts).
+            // If the specified version exists, use it; otherwise, default to SNMP version 2c.
+            version: getKey(snmp.Version, monitor.snmpVersion) || snmp.Version2c,
```

Comment:
How can it not exist?
We controll both the snmp library version (=> what we support there) and what `monitor.snmpVersion` is.

This looks like a really horrible hack
(I would have to dig into the code deeper to know what this does)

=> please remove the usage of `getKey`

Reply:
Resolved with c68b1c6274065eac57bb277d1151196792974951 and 0384b34007df05f6da97222c83e351c9b1d11607